aule/visitors/alfaASTVisitor.py

```diff
@@ -48,43 +44,64 @@ def visitArrayExpression(self, ctx):
     def visitBinaryOperator(self, ctx):
         return ctx.getText()
 
+    def visitSetOperator(self, ctx):
+        return ctx.getText()
+
+    def visitLogicalOperator(self, ctx):
+        return ctx.getText()
+
     def visitTargetStatement(self, ctx):
         clauses = []
         i = 0
         while ctx.targetExpression(i):
             clauses.append(self.visit(ctx.targetExpression(i)))
             i += 1
-        
         return TargetStatement(clauses)
 
-    def visitConditionStatement(self, ctx):
-        return self.visit(ctx.conditionExpression())
-
-    def visitTargetAndExpression(self, ctx):
+    def visitTargetLogicalExpression(self, ctx):
         left = self.visit(ctx.targetExpression(0))
         right = self.visit(ctx.targetExpression(1))
-        return LogicalExpression("AND", left, right)
-
-    def visitTargetOrExpression(self, ctx):
-        left = self.visit(ctx.targetExpression(0))
-        right = self.visit(ctx.targetExpression(1))
-        return LogicalExpression("OR", left, right)
+        operator = ctx.logicalOperator().getText()
+        return LogicalExpression(operator, left, right)
 
     def visitTargetUnaryExpression(self, ctx):
         left = self.visit(ctx.targetExpression())
         return UnaryExpression("NOT", left)
 
-    def visitTargetBaseExpression(self, ctx):
-        left = self.visit(ctx.targetExpression(0))
-        right = self.visit(ctx.targetExpression(1))
+    def visitBooleanExpression(self, ctx):
+        return LiteralBoolean(ctx.getText())
```

Comment:
Same here. I Suppose we can get bunch of quite tricky boolean literals w/o `.lower()`

Reply:
Done